src/source/core/Tasks.bs

```diff
@@ -364,6 +380,26 @@ namespace mc.tasks
     end if
   end function
 
+  function onTaskStateChange(state as string, task as mc.types.node)
+    if state = "done" or state = "stop"
```

Comment:
for more accuracy. let's do something like this.

```
isRunning = state = "run"
wasRunning = task._mIsRunning = true
if not wasRunning and state = "run"
  mc.setOrAddNodeField(task, "_mIsRunning", true)
else if wasRunning and state = "stop" or state = "done"
  mc.setOrAddNodeField(task, "_mIsRunning", false)
end if 

if wasRunning <> (task._mIsRunning = true)
  'DO TRACKING HERE - because we can be sure of the tasks's state.
end if

'DONT REMOVE THE OBSERVER - thought about this - we might reuse the task (it can happen sometimes)

```

Reply:
updated pr